src/cpp/src/visual_language/pipeline.cpp

```diff
@@ -99,8 +99,15 @@ class ov::genai::VLMPipeline::VLMPipelineImpl {
         size_t request_id = 0;
         size_t block_size = 1; // not used
         bool enable_prefix_caching = false;
+
+        auto tokenized_chat_history = m_inputs_embedder->get_tokenized_chat_history();
         size_t history_size = m_language.get_tensor("attention_mask").get_shape().at(1);
         size_t inputs_embeds_size = inputs_embeds.get_shape().at(1);
+
+        bool trust_encoded_history = m_inputs_embedder->is_trust_encoded_history();
+        if (!trust_encoded_history && (history_size + inputs_embeds_size) <= tokenized_chat_history.size())
```

Comment:
why do we need such condition `(history_size + inputs_embeds_size) <= tokenized_chat_history.size()` ?

Reply:
moved to checking of difference between histories only